### PR TITLE
Add command line tools for roboflow-python

### PR DESCRIPTION
**Pull request by tonylampada**
# Description

The main change introduce in this PR is the new `roboflowpy` command line tool to start making the api methods available from the command line.

CleanShot 2023-11-28 at 17 25 11

This PR also refactors some of the internal code moving towards a cleaner separation of concerns, moving the actual http calls to the `rfapi.py` adapter.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

## How has this change been tested, please provide a testcase or example of how you tested the change?

Tested on staging and production

## Docs

- [x] Need to document the new features both here in the python package and the API, but it's non-blocking. I'll be working on both next.

### CODE REVIEW

**Comment by capjamesg** [comment]
I am excited to review this, @tonylampada! Is this going to replace our JavaScript-based CLI

**Comment by capjamesg** [comment]
@tonylampada How do I install this?

**Comment by tonylampada** [comment]
> @tonylampada How do I install this?

@capjamesg `pip install -e .` from the root dir

**Comment by capjamesg** [comment]
I get the following error with the `roboflow download` command:

```
(base) james@jamesmac roboflow-python % roboflow download https://universe.roboflow.com/gdit/aerial-airport/dataset/1
Traceback (most recent call last):
  File "/Users/james/opt/anaconda3/bin/roboflow", line 8, in <module>
    sys.exit(main())
  File "/Users/james/Documents/GitHub/roboflow-python/roboflow/roboflowpy.py", line 181, in main
    args.func(args)
  File "/Users/james/Documents/GitHub/roboflow-python/roboflow/roboflowpy.py", line 14, in download
    w, p, v = args.datasetUrl.split("/")
ValueError: too many values to unpack (expected 3)
```

**Comment by tonylampada** [comment]
Sorry, lemme fix that

**Comment by tonylampada** [comment]
@capjamesg in the mean time you can probably do something like

`roboflow download gdit/aerial-airport/1 -f voc`

**Comment by capjamesg** [comment]
It would be nice to be able to pull from URLs, too. You can use an algorithm like:

1. Check if a string is a URL.
2. If the string is a URL, extract the path. Extract the org name, dataset name, and version.
3. If the string is a org name, dataset name, and version, extract the information as you do now.

**Comment by tonylampada** [comment]
It's fixed, that command is going to work now :-)

**Comment by capjamesg** [comment]
It works! Thank you!

**Review by capjamesg — APPROVED** [review]
One quick comment, then this looks good to go!

File: roboflow/roboflowpy.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        split=args.split,
+        num_retry_uploads=args.num_retries,
+        batch_name=args.batch,
+        tag_names=args.tag_names.split(",") if args.tag_names else [],
+        is_prediction=args.is_prediction,
+    )
+
+
+def _argparser():
+    parser = argparse.ArgumentParser(description="main description")
```
[the remark this answers is not in the document]
Comment (tonylampada) [reply roboflow/roboflowpy.py]: Woops, sorry about that. It's there now!

**Pull request merged**